src/thor/bidirectional_astar.cc

```diff
@@ -562,6 +564,9 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
   auto forward_time_info = TimeInfo::make(origin, graphreader, &tz_cache_);
   auto reverse_time_info = TimeInfo::make(destination, graphreader, &tz_cache_);
 
+  baldr::graph_tile_ptr origin_potential_tile = nullptr;
+  baldr::graph_tile_ptr destination_potential_tile = nullptr;
```

Comment:
I want to try one more thing here to squeeze every last drop of performance out of this. Here we only cache whatever tile we are using on a particular iteration, that is already very good but remember that we have multiple levels of tiles. every time the iteration jumps to a different heirarchy level we get a cache miss. we can avoid this cache miss as well. I'll push up a commit that adds this extra optimization later this afternoon!